### ISSUE TITLE
Node/app after condition stucks in executing

### ISSUE THREAD

**Issue by therealkefin** [frikky/Shuffle#218]
Data is sent from TheHive Webhook to Shuffle. Webhook sends data to Testing node (Repeat back to me). Afterwards data is sent to another app. In the most constellations the app seems to start, but stuck in executing and never finish...

Steps to reproduce:
1. Receive data from TheHive webhook (e.g. created observable)
2. Send data from Webhook to Testing_Node (Repeat back to me)
3. Set condition to $exec.operation=Creation and $exec.objectType=case_artifact between Testing_Node and any following app
4. Condition is obviously checked but any following app stucks in "executing status"

Same workflow with case creation in TheHive and condition with $exec.objectType=case strangely works...

Similar behaviour with other conditions when one value is $exec or similar...

Is this a known behaviour or is there a workaround?

**Comment by frikky** [comment]
Hey and thanks for the report! This sounds like a bug. I'll do some manual testing for it :+1: 

If possible, could you export the workflow you're building? (Go to the dashboard -> three dots in workflow -> export workflow)

**Comment by frikky** [comment]
Checked and I did indeed find a bug with the webhook. This whole thing works perfectly fine when using the "Execution Argument" field on the bottom. I'll send you a workflow later when I have time to fix it so you can test as well :)

**Comment by frikky** [comment]
Hey! Could you try downloading the new app sdk -> updating your apps and running it again? I pushed a bugfix to the App SDK which I think may have caused the issue.

1. Get SDK:
```
docker pull frikky/shuffle:app_sdk
```

2. Click "force update apps" in the /apps view.
![image](https://user-images.githubusercontent.com/5719530/101879477-ae918080-3b91-11eb-85a3-fdfb1f55f4d1.png)

**Comment by therealkefin** [comment]
Wow... just impressed with your speed. I pulled the new app sdk and gave it a try. It seems to work perfect now. Thanks for your amazing work.

**Comment by therealkefin** [comment]
Sorry it's me again. After playing around with the workflow it seems that it doesn't work again. I can't even reproduce my working flow from 1 hour before....

Same thing as before... Webhook (e.g. observable creation) -> Repeat back to me -> Condition ($exec.operation=Creation) -> any other node stucks in executing... 

I'm not sure why it worked this morning. As before everything works fine without conditions.

**Comment by frikky** [comment]
Hey again!

Let's dig a bit deeper then :)
What version of Shuffle Backend are you on? There was a database issue that made nodes stuck before 0.8.3 (check docker-compose :100: ) 
![image](https://user-images.githubusercontent.com/5719530/101973056-72aff700-3c35-11eb-9b4a-42ab10011fdd.png)


To properly fix it, I would need some help with debugging, as this is an App SDK issue, which are harder to fix. Full documentation for debugging are below, but here's the jist. And yes, we will make this way easier in the future :)
1. Run a workflow (make webhook trigger / click play button)
2. Check Orborus logs for the latest workflow running. Copy the ID you find there (this is the worker)
```
docker logs shuffle-orborus
```
![image](https://user-images.githubusercontent.com/5719530/101973084-ae4ac100-3c35-11eb-8f20-ea534f1bbbed.png)
3. Check the Worker logs (take the long ID from the line in the picture above. Copy all the logs from this one into a file and send it (Worker).
```
docker logs b0b0b8f86e3d2cc6f2ea03264277bc0d57d7e7af8a50ba9093e5d1d06f98f457
```
4. Find the App / Action in the logs you're looking for. The name will below a long ID. Run docker logs for this one as well, and send them
```
docker logs 5118a62fca7b417db7dbcde54cfd2b848d990433858dd3688c0600be70f8270c
```

![image](https://user-images.githubusercontent.com/5719530/101973130-008be200-3c36-11eb-9ab9-d528370447a5.png)

Here's more documentation on it: https://shuffler.io/docs/configuration#debugging

**Comment by therealkefin** [comment]
I don't get it. I didn't change anything since yesterday. I just ran the same workflow as yesterday again and it works... I tried to build even more complex workflows with multiple conditions and they all seem to work. How ist this possible? I will continue the work in the next days and try different things. If the workflow stucks again I will send you the logs.

However, my docker-compose doesn't contain a version number for the backend app.

<img width="153" alt="backend" src="https://user-images.githubusercontent.com/75784756/101981368-9fcbcc00-3c6c-11eb-947f-4bbcb0407b78.PNG">

Is this a problem?

**Comment by frikky** [comment]
These kinds of platforms are super complex, and there can be inconsistencies in how workflows run. This was fixed since the old models (less than 0.8.0). If there is no version number, you're running < 0.8.0, which means the first thing I'd suggest is to [updating Shuffle](https://shuffler.io/docs/configuration#updating_shuffle)

```
docker-compose down
git pull
docker-compose pull
docker-compose up -d
docker pull frikky/shuffle:app_sdk
docker pull ghcr.io/frikky/shuffle-worker:0.8.3
```

**Comment by frikky** [comment]
Closing as it's been more than a month since any activity. Just write here if you have an update :)
